### ISSUE TITLE
Creating multiple Databricks Workspace private endpoints concurrently erroring

### ISSUE THREAD

**Issue by q-thomaseastgate** [hashicorp/terraform-provider-azurerm#26373]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.4.7

### AzureRM Provider Version

3.108.0

### Affected Resource(s)/Data Source(s)

azurerm_private_endpoint

### Terraform Configuration Files

```hcl
resource "azurerm_private_endpoint" "databricks_private_endpoint" {
  name                = "pe-databricks"
  resource_group_name = <resource_group_name>
  location            = <location>
  subnet_id           = <databricks_private_endpoint>.id

  private_service_connection {
    name                           = "${local.deployment_id}-psc"
    is_manual_connection           = false
    private_connection_resource_id = azurerm_databricks_workspace.databricks_workspace.id
    subresource_names              = ["databricks_ui_api"]
  }

  # Add DNS A record to map connection to workspace
  private_dns_zone_group {
    name                 = "private-dns-zone-uiapi"
    private_dns_zone_ids = [<private_databricks_dns_zone>.id]
  }
}

resource "azurerm_private_endpoint" "databricks_auth_endpoint" {
  name                = "auth-databricks"
  resource_group_name = <resource_group_name>
  location            = <location>
  subnet_id           = <databricks_private_endpoint>.id

  private_service_connection {
    name                           = "psc-auth"
    private_connection_resource_id = azurerm_databricks_workspace.databricks_workspace.id
    is_manual_connection           = false
    subresource_names              = ["browser_authentication"]
  }

  # Add DNS A record to map connection to web auth
  private_dns_zone_group {
    name                 = "private-dns-zone-auth"
    private_dns_zone_ids = [<private_databricks_dns_zone>.id]
  }
}
```

### Debug Output/Panic Output

```shell
API Response:
----[start]----
{"status":"Failed","error":{"code":"ConcurrentUpdateError","message":"Call to Microsoft.Databricks/workspaces failed. Error message: Workspace update could not be completed because it has been updated by another process.","details":[]}}
-----[end]-----
```

### Expected Behaviour

The 2 Private links should have correctly created

### Actual Behaviour

The Databricks workspace would only allow the creation of 1 private endpoint at a time, so when the concurrency lock was removed by https://github.com/hashicorp/terraform-provider-azurerm/issues/26005, creating 2 Private endpoints for a Databricks workspace in parallel started to fail.

### Steps to Reproduce

1. Create a Databricks Workspace in Terraform
2. Create 2 Private Endpoints attaching to the Databricks Workspace

### References

https://github.com/hashicorp/terraform-provider-azurerm/pull/26006

**Comment by dghubble** [comment]
Has creating different private links in parallel caused a problem? Or is the issue that the private links should have created correctly in general (independent of serial vs parallel)?

**Comment by tombuildsstuff** [comment]
@lonegunmanb can you take a look into this one?

**Comment by adam-wynne** [comment]
I am getting the same behavior using Bicep, not terraform.  When I run the template it a second time, it works.

**Comment by gerrytan** [comment]
@q-thomaseastgate I was able to reproduce the problem and managed to work around it by using `depends_on` between the two resources.

From my internal discussion with @lonegunmanb This issue has resemblance to https://github.com/hashicorp/terraform-provider-azurerm/issues/16182.

At this point I'm not sure if we should fix this in azurerm by implementing a worker pool to limit the concurrent update, or in the ARM REST API. Doing the fix in the REST API will solve the problem in Bicep, CLI and other consumers. I will do some research and get back on this topic.

Meanwhile can you please check if you can workaround the problem using `depends_on` ?

**Comment by tandasima** [comment]
I don't think this is a TF issue per se. Databricks by design does not allow changes to happen at the same time. You need to make one change at a time. To fix this you need to add the following to your second endpoint so that the second endpoint is only created after the first one is complete:

```
# you need this to avoid the error you are seeing
  depends_on = [
    azurerm_private_endpoint.<first-endpoint-name>,  # sequence endpoints <------ need this
    azurerm_databricks_workspace.<workspace-name># explicit dependency <------ good to have
  ]

# this is good to avoid downtime if you are making this change in production. 
lifecycle {
    create_before_destroy = true
  }
```
